src/interface/control_scriptbox.cpp

```diff
@@ -0,0 +1,496 @@
+  if (m_suggestions_rect.contains(mouse_pos))
+  {
+    MouseCursor::current()->set_state(MouseCursorState::LINK);
+    m_selected_suggestion = std::min(static_cast<size_t>((mouse_pos.y - m_suggestions_rect.get_top() + m_suggestions_offset) /
+                                                         Resources::control_font->get_height()),
+                                     m_suggestions.size() - 1);
+    update_description();
+    return true;
+  }
+  else if (m_suggestion_description_rect.contains(mouse_pos))
+  {
+    for (const auto& parameter : m_function_parameters)
+    {
+      if (parameter.first.contains(mouse_pos))
+      {
+        m_hovered_parameter = &parameter;
+        break;
+      }
+    }
+    return true;
+  }
+
+  return ControlTextbox::on_mouse_motion(motion);
+}
+
+bool
+ControlScriptbox::on_mouse_wheel(const SDL_MouseWheelEvent& wheel)
+{
+  if (!m_has_focus)
+    return false;
+
+  if (m_suggestions.empty())
+    return ControlTextbox::on_mouse_wheel(wheel);
+
+  return m_suggestions_scrollbar->on_mouse_wheel(wheel);
+}
+
+bool
+ControlScriptbox::on_key_down(const SDL_KeyboardEvent& key)
+{
+  if (m_suggestions.empty())
+    return ControlTextbox::on_key_down(key);
+
+  switch (key.keysym.sym)
+  {
+    case SDLK_UP:
+    {
+      if (m_selected_suggestion > 0)
+      {
+        m_selected_suggestion--;
+
+        const float line_pos = m_selected_suggestion * Resources::control_font->get_height() + TEXT_Y_OFFSET;
+        if (line_pos - m_suggestions_offset < 0.f)
+          m_suggestions_offset = line_pos;
+
+        update_description();
+        return true;
+      }
+      else
+      {
+        m_suggestions.clear();
+      }
+      break;
+    }
+
+    case SDLK_DOWN:
+    {
+      if (m_selected_suggestion < m_suggestions.size() - 1)
+      {
+        m_selected_suggestion++;
+
+        const float line_pos = (m_selected_suggestion + 1) * Resources::control_font->get_height() + TEXT_Y_OFFSET;
+        if (line_pos - m_suggestions_offset > m_suggestions_rect.get_height())
+          m_suggestions_offset = line_pos - m_suggestions_rect.get_height();
+
+        update_description();
+        return true;
+      }
+      else
+      {
+        m_suggestions.clear();
+      }
+      break;
+    }
+
+    case SDLK_TAB:
+    case SDLK_RETURN: // ENTER
+    {
+      autocomplete();
+      return true;
+    }
+  }
+
+  return ControlTextbox::on_key_down(key);
+}
+
+void
+ControlScriptbox::update_description()
+{
+  m_suggestion_description.clear();
+  m_function_parameters.clear();
+  m_hovered_parameter = nullptr;
+
+  if (m_suggestions.empty())
+    return;
+
+  const squirrel::Suggestion& suggestion = m_suggestions.at(m_selected_suggestion);
+  if (!suggestion.reference) return;
+
+  const float max_width = (static_cast<float>(SCREEN_WIDTH) - m_suggestions_rect.get_right() -
+                           SUGGESTION_DESCRIPTION_RECT_SPACING);
+
+  using namespace squirrel;
```

Comment:
Move this to the top please. Then change `squirrel::Suggestion` case.

Also TBH it would be way more readable if we did `using sq = squirrel;` if you want to save some keystrokes without axing readability